### PR TITLE
IMX258 fixes for Vision Components modules

### PR DESCRIPTION
**Pull request by 6by9**
Adds alternate compatible string to adjust register set.

### CODE REVIEW

File: Documentation/devicetree/bindings/media/i2c/imx258.yaml [hunk 1]
```diff
@@ -14,10 +14,13 @@ description: |-
   type stacked image sensor with a square pixel array of size 4208 x 3120. It
   is programmable through I2C interface.  Image data is sent through MIPI
   CSI-2.
+  There are a number of variants of the sensor which can not be detected at
```
Comment (pelwell) [inline Documentation/devicetree/bindings/media/i2c/imx258.yaml]: "cannot" is preferred (ducks...)

File: drivers/media/i2c/imx258.c [hunk 2]
```diff
@@ -1782,6 +1817,19 @@ static int imx258_probe(struct i2c_client *client)
 
 	imx258->csi2_flags = ep.bus.mipi_csi2.flags;
 
+#ifdef CONFIG_ACPI
```
Comment (pelwell) [inline drivers/media/i2c/imx258.c]: Is this still necessary? i2c.h already provides a null (and NULL-returning) implementation of `i2c_of_match_device` for the case that `CONFIG_OF` is not enabled.

File: Documentation/devicetree/bindings/media/i2c/imx258.yaml (hunk 1 above)
Comment (6by9) [reply Documentation/devicetree/bindings/media/i2c/imx258.yaml]: Done

File: drivers/media/i2c/imx258.c (hunk 2 above)
Comment (6by9) [reply drivers/media/i2c/imx258.c]: Fixed. I'd initially used of_match_device, but then swapped.

**Pull request merged**